[Mod] SVG ファイルの全体のサイズが 75%に拡大されいたものを変更

manzyun opened the pull request:
Inkscape 1.0.1 にて確認と編集。

# 発端

`ergodash-rev1-2.svg` を元に [遊舎工房](https://yushakobo.jp/)様へのレーザー加工機に取り込ませるSVGファイルをInkscapeで作成しました。

入稿した際に、遊舎工房様から、

> データを拝見した所、通常のErgoDashよりサイズが小さいようです。
> （スイッチ穴の辺は14mm前後になるはずです）
> ご確認頂ければと思います。

とご指摘を受けたため、`ergodash-rev1-2.ai` をInkscapeで開いての確認ではありますが、
ページサイズが `ergodash-rev1-2.svg` と比較して以下の様になっておりました。

* `ergodash-rev1-2.ai`: 300mm辺
* `ergodash-rev1-2.svg` : 225mm辺

しかし、各オブジェクトの配置場所は、
「各スイッチ穴の幅間隔を見比べる限りでは違和感を抱かない」
ため、相対でファイル全体が75%に拡大(縮小)されたものと見受けて編集させていただきました。

manzyun committed:
[Mod] SVG ファイルの全体のサイズが 75%に拡大されいたものを変更

Inkscape 1.0.1 にて確認と編集。

takashiski commented on Case/rev1.2/ergodash-rev1-2.svg:
```diff
@@ -1,272 +1,751 @@
-</g>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg
+   xmlns:dc="http://purl.org/dc/elements/1.1/"
+   xmlns:cc="http://creativecommons.org/ns#"
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   version="1.1"
+   id="レイヤー_1"
+   x="0px"
+   y="0px"
+   viewBox="0 0 1133.8583 1133.8583"
+   xml:space="preserve"
+   width="300mm"
+   height="300mm"><metadata
```
拡縮の是非はわかりませんが、元のSVGにはなかったドキュメントサイズの単位が追加されたのでよさそうです。

下記リンクによると、単位がない場合はアプリケーションごとに独自にDPI解釈をするようです。

https://www.smartdiys.com/manual/smart-laser-mini-make-svg-file-new/
https://discordapp.com/channels/698792260095180820/734033481931292732/767274404303339520
```

manzyun replied:
> 元のSVGにはなかったドキュメントサイズの単位が追加されたのでよさそうです。

やや私事ですが、このコミットについての改善案として、
「ファイルのプロパティやその中のオブジェクトに対しても単位追加をした方が良いのかも」
と、懸案とその改善案が私個人の中で立案しましたので、失礼かもしれませんがここに書き残させて頂きます。

takashiski commented:
また別の話として、DXF形式ならSVGのようにアプリケーションごとに解釈がずれるということがないのでより望ましいかもしれません。

manzyun commented:
> また別の話として、DXF形式ならSVGのようにアプリケーションごとに解釈がずれるということがないのでより望ましいかもしれません。

それは存じ上げておりませんでした。勉強不足で申し訳ないです。

次回からの発注・入稿、及び私個人がファブスペースでレーザー加工機を利用する際の参考にさせて頂きます。

manzyun committed:
[Mod] キースイッチ穴が 14.1mm 辺になる様に全体を拡大。

ai ファイル上で適切な大きさを確認。

また、svg ファイル上で「キースイッチ穴」の一辺が 14.1mm 前後に成る倍率で、
拡大変形属性を各グルーブに追加。

ただし、Ubuntu 上の Inkscape ver.1.0 での確認と編集。

manzyun committed:
[Mod] `transform` 属性の値を `matrix` から `scale` へ変更

`matrix` (変形行列)指定よりも `scale` (拡大・縮小) 指定の方が、
今回の改良案に対しては適切と感じたため。

manzyun commented:
[先日のディスカッションコメント](https://github.com/omkbd/ErgoDash/pull/37#discussion_r507025941)の;

> やや私事ですが、このコミットについての改善案として、
> 「ファイルのプロパティやその中のオブジェクトに対しても単位追加をした方が良いのかも」
> と、懸案とその改善案が私個人の中で立案しました

上述通りの対応はSVGの仕様やInkscapeの仕様により現実的では無かったため、以下の対応を致しましたので連絡します。

# 実際の対応内容

各グループ要素に対して、
キースイッチ穴の一辺長を基準とし、一辺が 14.1mm から 14.2mm に成る様に
`transform="scale(1.327)"` と属性と値を書き加えました。

## 対応の詳細
2つコミットが在りますので、それぞれのコミットの解説をさせて頂きます。

### ebf91f2 について
1. Inkscapeで現在のキースイッチ穴辺のmmの値を確認
    * 約10.697mm
    * ![確認時のInkscapeのスクリーンショット。キースイッチ穴を選択した際に、メニューバー下の『幅』『高さ』の値が10.697mmと表示されている。](https://user-images.githubusercontent.com/461212/98079509-41fed400-1eb7-11eb-9c91-554f73dd314a.jpg)
2. 14.200mm前後の大きさに拡大したいので、その倍率の算出として `14.200 ÷ 10.697` を行う。結果、 `1.327474993` という倍率を得る。しかし小数点以降の桁数が多いため、 `1.3274` に切り捨て。
3. Inkscapeの「変形」機能を利用し、ファイル内全てのグループオブジェクトを選択して拡大。
    * `1.3274` のパーセンテージ変換は単純に10の2乗を行い算出の上、Inkscapeの変形ダイアログに入力。
    * ![拡大処理後のInkscapeのスクリーンショット。キースイッチ穴を選択した際に、メニューバー下の『幅』『高さ』の値が14.199mmと表示されている。](https://user-images.githubusercontent.com/461212/98080217-6effb680-1eb8-11eb-9e70-2454e14424c2.jpg)

この後、Inkscapeのメタデータを省いたSVGファイルとして保存しました。

### a972c51 について
コミットのdiffをご覧頂けると一目瞭然とは存じますが、SVGの仕様では `transform` 属性には幾つかの変換関数が使用できます。
[transform - SVG:Scalable Vector Graphics | MDN](https://developer.mozilla.org/ja/docs/Web/SVG/Attribute/transform)

今回のInkscapeの編集直後では `transform` 属性に `matrix`(変形行列)を指定されており、今回の改良の目的、
「SVGファイル全体のサイズが75%に拡大(縮小)されていたものを変更」
に対してはやや情報が多いと感想を抱いたため、 `scale` 変換関数に置換しました。

もちろん、Inkscapeでのファイルの正常読み取りは確認しました。
問題無く、期待どおりに拡大されて読み込みできました。

manzyun commented:
連絡が遅くなってしまいました。すみません。

私のフォークリポジトリ `manzyun/ErgoDash` に本流( `omkbd/ErgoDash` )のリポジトリを取り込んで、手元のInkscape( `Inkscape 1.1 (c68e22c387, 2021-05-23)` )で以下を確認しました：

- キースイッチ穴の辺が14mmであるか(mm未満は切り捨て)
- `ergodash-rev1-2.svg` ファイルの『ページサイズ』が300mmであるか(mm未満は切り捨て)

無事(？)、ミリメートル以上のズレは無い事が確認できました。

つきましては、このプルリクエストはマージされたものと解釈した上でクローズさせて頂きます。
ご対応有り難うございました。
